node/src/manager/commands/poi.rs

```diff
@@ -0,0 +1,655 @@
+    csv_writer_callcache.write_record(&[
+        "id",
+        "return_value",
+        "contract_address",
+        "block_number",
+    ])?;
+
+    for cc in &callcache_records {
+        csv_writer_callcache.write_record(&[
+            &cc.id,
+            &cc.return_value,
+            &cc.contract_address,
+            &cc.block_number.to_string(),
+        ])?;
+    }
+    csv_writer_callcache.flush()?;
+    return Ok(());
+}
+
+#[derive(Queryable, QueryableByName)]
+pub struct TableName {
+    #[sql_type = "Text"]
+    tablename: String,
+}
+
+/// Get all tables present in the subgraph schema. Will be used for dumping data
+fn get_subgraph_tables(
+    conn: &PooledConnection<ConnectionManager<PgConnection>>,
+    subgraph_schema_name: &str,
+) -> Result<Vec<TableName>, anyhow::Error> {
+    let query = format!(
+        "select tablename from pg_tables where schemaname='{}'",
+        subgraph_schema_name
+    );
+
+    let subgraph_tables = sql_query(query).get_results::<TableName>(conn).unwrap();
+    return Ok(subgraph_tables);
+}
+
+#[derive(Queryable, QueryableByName)]
+pub struct ColumnNameAndType {
+    #[sql_type = "Text"]
+    column_name: String,
+    #[sql_type = "Text"]
+    data_type: String,
+}
+
+/// Gathers entries from entity tables that contain the divergent block if in closed interval
+/// Diesel isn't of any use here because this is dynamic.
+/// Also need to catch the columns for passing to csv.
+///
+/// Note: there may be no return values for a table filtered on a block. In this case, eagerly return
+/// Note: map null values to empty string;
+fn get_filtered_subgraph_table_rows(
+    conn: &PooledConnection<ConnectionManager<PgConnection>>,
+    raw_client: &mut PGClient,
+    table_name: &String,
+    schema_name: &str,
+    divergent_block: i32,
+) -> Result<(Vec<ColumnNameAndType>, Vec<Vec<String>>), anyhow::Error> {
+    // We won't know the type of the returned table rows at compile time.
+    // And you can't use a placeholder for a table name when substituting a query string.
+    // Need to format prior and use the raw postgres driver.
+
+    let column_query = format!(
+        "SELECT column_name,data_type FROM information_schema.columns WHERE table_schema = '{}'
+     AND table_name   = '{}'
+       ORDER by ordinal_position",
+        schema_name, table_name
+    );
+    // let columns = raw_client.query(column_query, &[&schema_name, &table_name])?;
+
+    let columns = sql_query(column_query)
+        .get_results::<ColumnNameAndType>(conn)
+        .unwrap();
+
+    // Reserved keywords in SQL appear as subgraph tables unfortunately. I circumvent this with appending underscores
+    let entity_query = format!("select * from {}.{} as {}__ WHERE ({}::int4  <@ {}__.block_range AND upper_inc({}__.block_range))\
+OR ({}::int4 = lower({}__.block_range))",schema_name, table_name, table_name,divergent_block, table_name, table_name,divergent_block,table_name);
```

Comment:
`block_range` is never inclusive. This therefore boils down to `lower(block_range) = $n` which will get you all entities that were created or updated at block `$n`. If you also need to find which entities were deleted at block `$n`, you'd also need to query with `upper(block_range) = $n`. I would make that something like `EntityFilter::ClampedAtBlock(BlockNumber)` (we refer to restricting the range of an entity version as 'clamping')